src/renderer/components/+namespaces/namespace.store.ts

```diff
@@ -1,47 +1,46 @@
-import { action, observable, reaction } from "mobx";
+import { action, comparer, observable, reaction } from "mobx";
 import { autobind, createStorage } from "../../utils";
 import { KubeObjectStore } from "../../kube-object.store";
 import { Namespace, namespacesApi } from "../../api/endpoints";
-import { IQueryParams, navigation, setQueryParams } from "../../navigation";
+import { createUrlParam } from "../../navigation";
 import { apiManager } from "../../api/api-manager";
 import { isAllowedResource } from "../../../common/rbac";
 import { getHostedCluster } from "../../../common/cluster-store";
 
+const storage = createStorage<string[]>("context_namespaces", []);
+
+export const namespaceUrlParam = createUrlParam<string[]>({
+  name: "namespaces",
+  isSystem: true,
+  multiValues: true,
+  get defaultValue() {
+    return storage.get();
+  }
+});
+
 @autobind()
 export class NamespaceStore extends KubeObjectStore<Namespace> {
   api = namespacesApi;
-  contextNs = observable.array<string>();
-
-  protected storage = createStorage<string[]>("context_ns", this.contextNs);
-
-  get initNamespaces() {
```

Comment:
Namespace store should not be exported to public API.

Reply:
Could you clarify why? It's already exported and extensions are going to need shared stores with Lens so that Lens does not end up doing multiple queries/watches to Kubernetes cluster.

Reply:
Same reason as you suggested to add facade for cluster-store and workloads-store..

Reply:
If this class must be changed so that it breaks the API then this PR needs to add a facade that keeps it compatible, right?

Reply:
I think we should now very carefully export everything to public API and if export required then provided some `facade`. Otherwise we're locking ourself more and more from not-changing apis.. That's a good start for legacy :)

Reply:
>> If this class must be changed so that it breaks the API then this PR needs to add a facade that keeps it compatible, right?
No, facade should be exported to public, so internal changing internal apis doesn't affect to public APIs.

Reply:
I meant: because there is no facade for this class then a PR that is going to break the API needs to add it. While doing so we probably will see if that's even possible to achieve. Anyway my original comment here was meant to indicate that this will change the public api, not only internals.

Reply:
Ok, if we gonna continue export namespace-store as-is it should have some facade.. or at least leave apis as is (i just provided better names in general, no logic changed)

Reply:
public apis reverted for the store